Modules/Devices/src/Devices.Application/PushNotifications/Commands/UpdateDeviceRegistration/Handler.cs

```diff
@@ -25,19 +25,23 @@ public Handler(IPushService pushService, IUserContext userContext)
         _activeDevice = userContext.GetDeviceId();
     }
 
-    public async Task<Unit> Handle(UpdateDeviceRegistrationCommand request, CancellationToken cancellationToken)
+    public async Task<UpdateDeviceRegistrationResponse> Handle(UpdateDeviceRegistrationCommand request, CancellationToken cancellationToken)
     {
         var parseHandleResult = PnsHandle.Parse(request.Handle, DeserializePlatform(request.Platform));
+        DevicePushIdentifier res;
         if (parseHandleResult.IsSuccess)
         {
-            await _pushService.UpdateRegistration(_activeIdentity, _activeDevice, parseHandleResult.Value, request.AppId, DeserializeEnvironment(request.Environment ?? PRODUCTION_ENVIRONMENT), cancellationToken);
+            res = await _pushService.UpdateRegistration(_activeIdentity, _activeDevice, parseHandleResult.Value, request.AppId, DeserializeEnvironment(request.Environment ?? PRODUCTION_ENVIRONMENT), cancellationToken);
         }
         else
         {
             throw new ApplicationException(new ApplicationError(parseHandleResult.Error.Code, parseHandleResult.Error.Message));
         }
 
-        return Unit.Value;
+        return new UpdateDeviceRegistrationResponse
+        {
+            DevicePushIdentifier = res
+        };
```

Comment:
In such a simple case I would do the mapping to the properties in the constructor. So just pass the identifier as a parameter.

```suggestion
        return new UpdateDeviceRegistrationResponse(res);
```